configs/README.md

```diff
@@ -15,7 +15,7 @@ A detailed description of the parameters for model configuration (i.e., initiali
 | forcing_resolution | double (scalar)| - | sec/min/hr | temporal resolution | - | timestep of the forcing data. Recommended value of 3600 seconds. |
 | endtime | double (scalar)| >0 | sec, min, hr, d | simulation duration | - | time at which model simulation ends |
 | layer_soil_type | int (1D array) | - | - | state variable | - | layer soil type (read from the database file soil_params_file) |
-| max_soil_types | int | >1 | - | - | - | maximum number of soil types read from the file soil_params_file (default is set to 15) |
+| max_valid_soil_types | int | >1 | - | - | - | maximum number of valid soil types read from the file soil_params_file (default is set to 12; model not valid for soil_type = waterbody, lava, glacier, etc.) |
```

Comment:
It should be noted that if this value is more than the constant defined here

https://github.com/NOAA-OWP/LGAR-C/blob/5120df42f6eb0dea32aef4659daaaa7abb1e53d8/include/all.hxx#L39

then an error/exit condition exists.  I don't think this the number impacts the implementation details of this PR, but it is a somewhat odd out-of-band validation that is relevant.

Reply:
I am not seeing if MAX_NUM_SOIL_TYPES is used anywhere, I am going to push a change to put this as an upper limit.

Reply:
I realized that it isn't used in any of the BMI code paths, for sure, but in the `src/main.cxx` there is this check:
https://github.com/NOAA-OWP/LGAR-C/blob/5120df42f6eb0dea32aef4659daaaa7abb1e53d8/src/main.cxx#L258